backend/server.py

```
@@ -17,16 +35,351 @@
+    try:
+        # Call the function to get current MCP enabled status
+        # This assumes you've implemented a way to check if MCP is enabled
+        result = {"enabled": True}  # Replace with actual status check
+        return result
+    except Exception as e:
+        return {"error": f"Error getting MCP status: {str(e)}"}
+
+
+# MCP Toggle Endpoint
+@app.post("/mcp/toggle")
+async def toggle_mcp(request: dict = {"enabled": True}):
```

Ruff (0.8.2)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)